docs/plugins/esm-external-require.md

```diff
@@ -4,36 +4,14 @@ The `esmExternalRequirePlugin` is a built-in Rolldown plugin that converts Commo
 
 ## Why This Is Needed
 
-When bundling code with Rolldown, external dependencies that use `require()` calls are not automatically converted to ESM imports. Although setting `platform: 'node'` can convert `require` calls into `import`, it does so by generating code like:
+When bundling code with Rolldown, `require()` calls for external dependencies are not automatically converted to ESM imports to preserve the semantics of `require()`. While Rolldown injects `require` function when `platform: 'node'` is set, it does so by generating code like:
```

Comment:
- "external dependencies that use `require()` calls" felt like `require` calls inside external dependencies
- clarified the reason why rolldown keeps `require` by default